code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,278 @@
+		/obj/effect/decal/cleanable/ash = "[initial(name)] with INCREDIBLY IMPORTANT ARTIFACT- DELIVER TO SCIENCE DIVISION. HANDLE WITH CARE.",
+	)
+
+	color = pick(department_colors) //eh, who gives a shit.
+	name = special_name ? junk_names[junk] : "important [initial(name)]"
+
+	junk = new junk(src)
+	return TRUE
+
+/// Subtype that's always junkmail
+/obj/item/mail/junkmail/Initialize()
+	..()
+	junk_mail()
+
+/// Crate for mail from CentCom.
+/obj/structure/closet/crate/mail
+	name = "mail crate"
+	desc = "A certified post crate from CentCom."
+	icon_state = "mail"
+
+/// Crate for mail that automatically generates a lot of mail. Usually only normal mail, but on lowpop it may end up just being junk.
+/obj/structure/closet/crate/mail/full
+	name = "brimming mail crate"
+	desc = "A certified post crate from CentCom. Looks stuffed to the gills."
+
+/obj/structure/closet/crate/mail/update_icon_state()
+	. = ..()
+	if(opened)
+		icon_state = "[initial(icon_state)]open"
+		for(var/obj/item/mail/Mail in src)
+			icon_state = initial(icon_state)
+			break
+	else
+		icon_state = "[initial(icon_state)]sealed"
+
+/obj/structure/closet/crate/mail/full/Initialize()
+	. = ..()
+	var/list/mail_recipients = list()
+	for(var/mob/living/carbon/human/alive in GLOB.player_list)
+		if(alive.stat != DEAD)
+			mail_recipients += alive
+	for(var/iterator in 1 to 22)
+		var/obj/item/mail/New_mail
+		if(prob(70))
+			New_mail = new /obj/item/mail(src)
+		else
+			New_mail = new /obj/item/mail/envelope(src)
+		var/mob/living/carbon/human/mail_to
+		if(mail_recipients.len)
+			mail_to = pick(mail_recipients)
+		if(prob(50)) //so after 21 passes if everyone's at least gotten something we'll junkmail it up
+			mail_recipients -= mail_to
+		if(mail_to)
+			New_mail.initialize_for_recipient(mail_to)
+		else
+			New_mail.junk_mail()
+
+/// Mailbag.
+/obj/item/storage/bag/mail
+	name = "mail bag"
+	desc = "A bag for letters, envelopes, and other postage."
+	icon = 'icons/obj/library.dmi'
+	icon_state = "bookbag"
+	worn_icon_state = "bookbag"
+	resistance_flags = FLAMMABLE
+
+/obj/item/storage/bag/mail/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_w_class = WEIGHT_CLASS_NORMAL
+	STR.max_combined_w_class = 42
+	STR.max_items = 21
+	STR.display_numerical_stacking = FALSE
+	STR.set_holdable(list(
+		/obj/item/mail,
+		/obj/item/small_delivery,
+		/obj/item/paper
+	))
+
+/obj/item/paper/fluff/junkmail_redpill
+	name = "smudged paper"
+	icon_state = "scrap"
+
+/obj/item/paper/fluff/junkmail_redpill/Initialize()
+	. = ..()
+	var/code = random_nukecode()
+	for(var/obj/machinery/nuclearbomb/selfdestruct/self_destruct in GLOB.nuke_list)
+		self_destruct.r_code = code
+	message_admins("Through junkmail, the self-destruct code was set to \"[code]\".")
+	info = "<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[code[rand(1,5)]][code[rand(1,5)]]...'"
```

Comment:
I'm only rationalizing this in my head as okay since the arcade pulse rifle is about the same in my eyes.

Reply:
Made into a 1-1000 rarity to have the true nuke codes as of ~3 commits ago, putting it on the level, in addition to the fact you're not getting all the code values, the rarity should now be on the level of pulse rifle rarity. Blame Coiax.